server/src/main/java/org/elasticsearch/indices/IndicesRequestCache.java

```diff
@@ -137,7 +139,9 @@ BytesReference getOrCompute(CacheEntity cacheEntity, Supplier<BytesReference> lo
      * @param cacheKey the cache key to invalidate
      */
     void invalidate(CacheEntity cacheEntity, DirectoryReader reader, BytesReference cacheKey) {
-        cache.invalidate(new Key(cacheEntity, reader.getVersion(), cacheKey));
+        if (reader.getReaderCacheHelper() != null) {
```

Comment:
Why is it possible to have a null cache helper here ?

Reply:
some reader wrappers will return null here if they change the live-docs of a reader for instance. But I wonder in what case that happens?

Reply:
I've replaced this check with an assertion as we should always have readerCacheKey.